### PR TITLE
Nanoforge 2.0

### PR DESCRIPTION
**Pull request by Evankhell561**
## About The Pull Request

the nanoforge is a machine capable of doing everything an autolathe does but wants an oddity with perk to work. You are also capable of creating compressed matter at the cost of all stored materials and you can produce your own designs at the cost of the oddity perk.

It can only be used by technomancers or those with high stats in MEC.

## Why It's Good For The Game
avoid suicides.

## Changelog
:cl:
add: the nanoforge.
/:cl:

### CODE REVIEW

**Comment by Evankhell561** [comment]
this was assigned to me due to #5522  having unexpected errors.

**Comment by 1glitchycent** [comment]
based speedmerge

File: code/game/machinery/autolathe/nanoforge.dm [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/obj/machinery/autolathe/nanoforge
+	name = "matter nanoforge"
+	desc = "It consumes items and produces compressed matter."
+	icon_state = "nanoforge"
+	icon = 'icons/obj/machines/autolathe.dmi'
+	use_oddities = TRUE
+	is_nanoforge = TRUE
+	circuit = /obj/item/weapon/electronics/circuitboard/nanoforge
+	var/list/nanoforge_designs = list()
+	var/list/tags_to_spawn = list(SPAWN_DESING)
```
Comment (drexample) [inline code/game/machinery/autolathe/nanoforge.dm:10]: With https://github.com/discordia-space/CEV-Eris/pull/5832 merged, change to SPAWN_DESIGN

**Review by drexample — CHANGES_REQUESTED** [review]

File: code/game/machinery/autolathe/nanoforge.dm (hunk 1 above)
Comment (Evankhell561) [reply code/game/machinery/autolathe/nanoforge.dm:10]: that's not true.
#5797

**Comment by MacBlaze1** [comment]
godspeed Evan

**Comment by MacBlaze1** [comment]
I like the nanoforge and what it does, but I feel that it keeps Technomancers too reliant on Guild for all their tools and needs. Moebius, NeoTheology and Guild are all independent of each other for producing designs. I feel this machine should make technomancers the same.

**Review by drexample — APPROVED** [review]

**Pull request merged**